department-of-veterans-affairs/vets-website
# Pension 80009 bank account pattern

## Summary
Update the 'Account information for direct deposit' page in the 'Additional information' chapter of the Pension Benefits form (21p-527EZ)

## Description
Replaces the deprecated [bank account widget](https://github.com/department-of-veterans-affairs/vets-website/blob/16a5c58e294709667a773a87b73e011174822e77/src/platform/forms-system/src/js/definitions/bankAccount.js) with a reusable [bankAccount web component pattern](https://github.com/department-of-veterans-affairs/vets-website/blob/f7a10d9aaf2ba65de452d8667b24200bff68fc5f/src/platform/forms-system/src/js/web-component-patterns/bankPattern.jsx) in Platform.

## Acceptance criteria
[Issue in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80009)

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`
4. Go to `http://localhost:3001/pension/application/527EZ` in your browser

## How to verify
1. Go to the direct deposit page in the 'Additional information' chapter
2. Select 'Yes' on the 'Do you have a bank account to use for direct deposit?' question
3. Continue to the 'Account information for direct deposit' page
4. Verify that the fields have been updated to use web components
5. Verify that the 'type', 'account number' and 'routing number' are required and validated
6. Verify that the form data is saved to the api and redux store
7. Verify that the 'Account information for direct deposit' data can be viewed and edited on the 'Review application' page
8. Verify that the form can be submitted

## Related issue(s)
[Issue in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80009)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/76760)

## Testing done

## Screenshots
| Before | After |
| ------ | ------ |
|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_additional-information_account-information](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/2a7ca154-9418-488a-8614-470ed4d82757)|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_form-saved](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/a28e9f8b-1a6f-40d9-9026-c77c9ef9498f)


## What areas of the site does it impact?
Pension Benefits Application

### Quality Assurance & Testing

### Error Handling


### Authentication
